Assets/Scripts/Layers/LayerTypes/LayerData.cs

```diff
@@ -11,16 +12,39 @@
 namespace Netherlands3D.Twin.Layers
 {
     [Serializable]
-    public class LayerData
+    public abstract class LayerData
     {
         [SerializeField, DataMember] protected Guid UUID = Guid.NewGuid();
         [SerializeField, DataMember] protected string name;
         [SerializeField, DataMember] protected bool activeSelf = true;
-        [SerializeField, DataMember] protected Color color = new Color(86f / 256f, 160f / 256f, 227f / 255f);
+        
+        /// <summary>
+        /// The default color of a layer.
+        /// 
+        /// This will influence how it is displayed in the layers side-panel, it does not automatically imply any
+        /// coloring in the styling of the layer but can be used to tell layers apart from one another in the layer
+        /// panel.
+        ///
+        /// Each type of layer could decide to use this value to influence the default styling by listening to the
+        /// ColorChanged event and applying the color to the relevant color field in the default Style, such as fill
+        /// for polygon vector layers, or stroke color for line polygon layers.
+        /// </summary>
+        [SerializeField, DataMember] protected Color color = new(86f / 256f, 160f / 256f, 227f / 255f);
+
         [SerializeField, DataMember] protected List<LayerData> children = new();
         [JsonIgnore] protected LayerData parent; //not serialized to avoid a circular reference
         [JsonIgnore] protected int rootIndex = -1;
         [SerializeField, DataMember] protected List<LayerPropertyData> layerProperties = new();
+        
+        /// <summary>
+        /// A list of styles with their names (which are meant as machine-readable names and not human-readable names,
+        /// for the latter the 'title' field exists), including a default style that always applies.
+        /// </summary>
+        [SerializeField, DataMember] protected Dictionary<string, LayerStyle> styles = new()
```

Comment:
gaat dit goed met serializen? ik meen me te herinneren dat in builds alleen Lists werken (in de editor werkt dit wel)